fitbenchmarking/fitbenchmarking
# 205 improve documentation on entry points

#### Description of Work

Fixes #205
Refactored do_benchmarking and edited docstrings to explain the code in a more understandable way.

Also contains commits for:
- Indentaion fixes
- Fix for broken mantid example
- Fix for expert example (I removed some pass through functions in a previous commit that were used by expert. I have put these back in and marked them as API functions.)

#### Testing Instructions

1. Check everything works as before
2. Read docstrings and see if it makes sense.
